Client/Application/SceneDelegate.swift

```diff
@@ -0,0 +1,308 @@
+        }
+
+    }
+
+    /// Use this method to notify the user that the specified activity couldn't be completed.
+    ///
+    /// We need to think about the right ways to notifiy users
+    func scene(
+        _ scene: UIScene,
+        didFailToContinueUserActivityWithType userActivityType: String, error: Error
+    ) {
+        // no-op
+    }
+
+    // MARK: - Responding to Scene (environment) Changes
+
+    /// Use this method to handle certain environment changes and adjust your scene accordingly.
+    ///
+    /// See how / what needs to be done here so that we can lock orientation on modal presentation as well.
+    func windowScene(
+        _ windowScene: UIWindowScene,
+        didUpdate previousCoordinateSpace: UICoordinateSpace,
+        interfaceOrientation previousInterfaceOrientation: UIInterfaceOrientation,
+        traitCollection previousTraitCollection: UITraitCollection
+    ) {
+        // no-op
+        // return self.orientationLock
+    }
+
+    // MARK: - Performing Tasks
+
+    /// Use this method to handle a selected shortcut action.
+    ///
+    /// Invoked when:
+    /// - a user activates the application by selecting a shortcut item on the home screen AND
+    /// - the window scene is already connected.
+    func windowScene(
+        _ windowScene: UIWindowScene,
+        performActionFor shortcutItem: UIApplicationShortcutItem,
+        completionHandler: @escaping (Bool) -> Void
+    ) {
+        let handledShortCutItem = QuickActions.sharedInstance.handleShortCutItem(shortcutItem,
+                                                                                 withBrowserViewController: self.browserViewController)
+
+        completionHandler(handledShortCutItem)
+    }
+
+    // MARK: - Misc. Helpers
+
+    private func configureWindowFor(_ scene: UIScene) -> UIWindow {
+        guard let windowScene = (scene as? UIWindowScene) else {
+            return UIWindow(frame: UIScreen.main.bounds)
+        }
+
+        let window = UIWindow(windowScene: windowScene)
+
+        if !LegacyThemeManager.instance.systemThemeIsOn {
+            window.overrideUserInterfaceStyle = LegacyThemeManager.instance.userInterfaceStyle
+        }
+
+        return window
+    }
+
+    private func configureRootViewController() -> UINavigationController {
+        let browserViewController = BrowserViewController(profile: profile, tabManager: tabManager)
+
+        // TODO: When we begin to support multiple scenes, remove this line and the reference to BVC in SceneDelegate.
+        self.browserViewController = browserViewController
+
+        let navigationController = UINavigationController(rootViewController: browserViewController)
+        navigationController.isNavigationBarHidden = true
+        navigationController.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
+
+        return navigationController
+    }
+
+}
+
+/**
+    TODO: We need to figure out what the best approach to addressing `foregroundBVC` is.
```

Comment:
These commented out extensions are one way I thought we could more gracefully handle `foregroundBVC`. Will look into more options and ask everyone about it first.

Reply:
Committing commented code is not the way to store this info. You can stash it locally or even create a shared doc if you want to share it with others before creating a PR.